### PR TITLE
Use new lineman buildpack

### PR DESCRIPTION
**Pull request by tejasbubane**
This will remove the need to keep generated css and js in version control.

Closes https://github.com/exercism/discussions/issues/28

Using [hone's buildpack](https://github.com/hone/lineman.js-on-Heroku) for cedar-14. Following all instructions in the readme there.

With these buildpacks:
```
⇒  heroku buildpacks
=== immense-dawn-75760 Buildpack URLs
1. https://github.com/heroku/heroku-buildpack-ruby.git
2. https://github.com/heroku/heroku-buildpack-nodejs
3. https://github.com/hone/heroku-buildpack-static
```

### CODE REVIEW

**Comment by tejasbubane** [comment]
Deployed to https://immense-dawn-75760.herokuapp.com/

I have a little problem with postgres setup on heroku. `heroku build` succeeds but application giving error (see stacktrace on application url). 

Ran migrations on heroku, but seeding data fails:

```
⇒  heroku run rake db:seed
Running rake db:seed on ⬢ immense-dawn-75760... up, run.4088
RuboCop's rake task could not be loaded
cannot load such file -- rubocop/rake_task
psql: could not connect to server: No such file or directory
	Is the server running locally and accepting
	connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?
```

I suspect missing seed data is the reason for that application error.

**Comment by mhelmetag** [comment]
I have that trouble as well. I can also run `db:migrate` and can use `bin/console` to manipulate models.

I did start walking down the path of configuring my heroku instance to run in development (by tweaking some configs) to see if `db:seed` is development/local only but... have been even less successful in that.

I also get the same error as seen on your machine when trying to login on mine (at least I get the index page though 😆 ):
```
undefined method `username' for #<Guest:0x007f5895b40d30> (NoMethodError)
```

**Comment by tejasbubane** [comment]
I did not even try to login 😞  and cannot see index page. Any other workaround to test this lineman thing?

File: config/database.yml [hunk 1]
```diff
@@ -17,4 +17,7 @@ development:
   database: exercism_development
 
 production:
+  adapter: postgresql
+  encoding: unicode
   url: <%= ENV["DATABASE_URL"] %>
+  database: exercism_production
```
Comment (tejasbubane) [inline config/database.yml]: These are some hacks I was trying to get the postgres working on heroku but it did not work. Will remove these.

**Comment by kytrinyx** [comment]
Huh. I am not sure what we would have to do to get the seeds running in production. How about skipping seeds and running db migrate and then having a few people log in?

**Comment by kgrz** [comment]
Got a chance to debug this after getting Heroku access from @tejasbubane. The problem seems to be missing username, password, host details in the production database config. This is not a problem for most commands since AR handles this part in other tasks (`db.rake`). In `seeds.rake`, we call in a system command which relies on username, password etc. since it's mostly intended to be run in dev mode.

You probably can fix this by changing:

```ruby
    system({ 'PGPASSWORD' => config.password },
           'psql', '-h', config.host, '-p', config.port,
                   '-U', config.username,
                   '-d', config.database, '-f', 'db/seeds.sql')
```

to

```ruby
    system('psql', ENV['DATABASE_URL'], '-f', 'db/seeds.sql')
```

And running `db:seeds:fetch` tasks before the `db:seeds` step since the `seeds.sql` won't be present on the file system initially.

**Comment by tejasbubane** [comment]
@kgrz Thanks a lot for looking into this. But I am still facing some issues with the seeds:

```shell
⇒  heroku run rake db:seeds:fetch
Running rake db:seeds:fetch on ⬢ immense-dawn-75760... up, run.2291
fetching over the wire

⇒  heroku run rake db:heroku_seed
Running rake db:heroku_seed on ⬢ immense-dawn-75760... up, run.5442
db/seeds.sql: No such file or directory

⇒  heroku run rake db:seeds:fetch && rake db:heroku_seed
Running rake db:seeds:fetch on ⬢ immense-dawn-75760... up, run.1066
fetching over the wire
No DATABASE_URL environment variable found. Did you add postgresql addon?
If you're trying to run this task on your local machine, prefer the `db:seed` task instead
```

Would like to know how you ran the seed task.

@kytrinyx I have finally made the heroku instance serve exercism. https://immense-dawn-75760.herokuapp.com/ The assets are compiled using the nodejs buildpack 🎉 

Only thing left is to make travis happy. I added `npm install` step to travis yml file but it does not seem to get `compass` command even though the gem is [not under any group](https://github.com/tejasbubane/exercism.io/blob/use-lineman-buildpack/Gemfile#L27) in Gemfile (which means it should be available in all environments). Can anyone help me fix that?

**Comment by kgrz** [comment]
@tejasbubane Try using `bundle exec compass`. Heroku is adding the binaries to the environment path, and travis might not do it automatically for all the binaries that get installed via `bundle install --path=vendor`

**Comment by kgrz** [comment]
> Would like to know how you ran the seed task.

My best estimate for this is that you ran the commands individually which means that heroku spun two dynos and the `seeds.sql` file that got saved in one dyno didn't get reflected in the other. I ran a bash prompt on heroku, and then ran both the commands one after the other. That is, didn't close the connection, and so was able to see the `seeds.sql` persisted to the file system.

We can probably solve this by making `heroku_seed` task (or perhaps both seed and heroku_seed) depend on the `seeds:fetch` task. Will open a PR for that.

**Comment by kytrinyx** [comment]
> Only thing left is to make travis happy.

Did `bundle exec compass` work @tejasbubane?

**Comment by tejasbubane** [comment]
@kytrinyx Yes, `bundle exec compass` worked, but now the `test:everything` is failing since it cannot find `lineman` for some reason. Not sure why since it was running previously. [Details here](https://travis-ci.org/exercism/exercism.io/builds/141548020)

**Comment by kytrinyx** [comment]
does it need to install lineman globally in order to have the command available? (I'm not familiar with node)

**Comment by tejasbubane** [comment]
@kytrinyx Installing lineman globally fixed the issue. 👍 
Test this and let me know if you want me to squash the commits before merging.

File: public/js/styleguide.js [hunk 2]
```diff
@@ -1,60 +0,0 @@
-// NOTICE!! DO NOT USE ANY OF THIS JAVASCRIPT
-// IT'S ALL JUST JUNK FOR OUR DOCS!
-// ++++++++++++++++++++++++++++++++++++++++++
```
Comment (tejasbubane) [inline public/js/styleguide.js]: This was not autogenerated, but because of this comment I removed it.

Comment (kytrinyx) [inline public/js/styleguide.js]: Nice!

**Comment by kytrinyx** [comment]
Holy smokes this is exciting.

I'm heading to a conference, and will take a look at this Thu/Fri next week.

**Comment by kytrinyx** [comment]
Update: I spent the past two days catching up on a bajillion emails. I'm going to see if I can get a good look at this in the next few days.

**Comment by tejasbubane** [comment]
@kytrinyx No worries, take your time. 
Meanwhile I am working on fingerprinting assets [as I had mentioned earlier here](https://github.com/exercism/exercism.io/issues/2757#issuecomment-183581134). I think I almost got it working for `lineman build` on my local machine using https://github.com/testdouble/grunt-asset-fingerprint. This should allow proper cache busting at different levels. Will send a separate PR for that.

**Comment by kytrinyx** [comment]
Update: I'm still snowed under and haven't had time to look at this. It's still on my list of important things to get to as soon as I can!

**Comment by kytrinyx** [comment]
Merged in bc25ed5b6aba399ed86eb3433ba0a8ff68409240

AND IT TOTALLY WORKS :tada: :tada: :tada:

Thank you so much @tejasbubane, and I'm so sorry it took me so long to actually deal with it. It was particularly scary to mess with the deploy, so I kept putting it off :(

I'm so relieved that this is finally taken care of.

**Comment by tejasbubane** [comment]
> Thank you so much @tejasbubane, and I'm so sorry it took me so long to actually deal with it. It was particularly scary to mess with the deploy, so I kept putting it off :(

No worries, I can understand :) Cheers! 🎉

**Pull request closed without merging**